File: Ontology/DataPropertyDomain.cs
namespace Ontology
{
    public class DataPropertyDomain:
        DataPropertyAxiom,
        IDataPropertyDomain
    {
        private readonly IClassExpression _domain;

        public DataPropertyDomain(
            IOntology               ontology,
            IDataPropertyExpression dataPropertyExpression,
            IClassExpression        domain
            ) : base(
                ontology,
                dataPropertyExpression)
        {
            _domain = domain;
        }

        IClassExpression IDataPropertyDomain.Domain => _domain;
    }
}

File: Web/ClientApp/src/app/Ontology/DataPropertyDomain.ts
import { DataPropertyAxiom } from "./DataPropertyAxiom";
import { IAxiomVisitor } from "./IAxiomVisitor";
import { IClassExpression } from "./IClassExpression";
import { IDataPropertyDomain } from "./IDataPropertyDomain";
import { IOntology } from "./IOntology";
import { IDataPropertyExpression } from "./IPropertyExpression";

export class DataPropertyDomain
    extends DataPropertyAxiom
    implements IDataPropertyDomain
{
    constructor(
        ontology                     : IOntology,
        public DataPropertyExpression: IDataPropertyExpression,
        public Domain                : IClassExpression
        )
    {
        super(ontology);
    }

    Accept(
        visitor: IAxiomVisitor
        )
    {
        visitor.DataPropertyDomain(this);
    }
}
